crates/core/tedge_api/src/mqtt_topics.rs

```diff
@@ -803,6 +803,14 @@ impl IdGenerator {
     pub fn is_generator_of(&self, cmd_id: &str) -> bool {
         cmd_id.contains(&self.prefix)
     }
+
+    /// Extract c8y's operation ID from the provided thin-edge command ID.
+    pub fn get_c8y_operation_id(&self, cmd_id: &str) -> Option<String> {
```

Comment:
issue(blocking): this won't always be c8y operation id, consider this test where it would return `1111`:

https://github.com/thin-edge/thin-edge.io/blob/e48ffd7518720aaef774fb9fc086714d6eca45a1/tests/RobotFramework/tests/cumulocity/configuration/configuration_operation.robot#L115-L124

As it's only relevant to c8y mapper, it should probably be moved there and we should make sure that it's run only on commands spawned by the mapper that indeed contain c8y operation id.

Reply:
thought: overall I think we're starting to put a lot of information as prefixes or suffixes into command ids:

- if they're run as part of sub-workflow (`sub:` prefix)
- what c8y operation id they're spawned in response to (`c8y_mapper_ext/src/converter.rs:628`)
- the mapper that spawned them (`prefix` field in `IdGenerator`)

And this is sort of in between two approaches:
- opaque identifiers that are created by some thing and then handled only by the same thing
- structured identifiers that contain multiple fields

I think 1st approach is what we should target and this current mix a result of c8y operations being decoded and commands being spawned in two totally different places, so it's hard to save this state except in a command id.

As in the 2nd paragraph of [this comment](https://github.com/thin-edge/thin-edge.io/pull/3032#issuecomment-2299191772) the situation should improve once handling incoming c8y operation messages is moved to a dedicated operation handling actor. Now I have bit better picture for this refactoring.

Reply:
I'll change this function to generic and move the check for the number to c8y-mapper.

```rust
    pub fn get_value(&self, cmd_id: &str) -> Option<String> {
        cmd_id
            .strip_prefix(&format!("{}-", self.prefix))
            .map(|s| s.to_string())
    }
```

Reply:
Then one more improvement would be to avoid allocations and just return a slice:

```rust
pub fn get_value<'a>(&self, cmd_id: &'a str) -> Option<&'a str> {
    cmd_id
        .strip_prefix(&self.prefix)
        .and_then(|s| s.strip_prefix('-'))
}
```

Reply:
Done in https://github.com/thin-edge/thin-edge.io/pull/3076/commits/0136b9ebe68a38f28380210b0cbb53e34f69beed